simphony/cuds/abc_modeling_engine.py

```diff
@@ -22,185 +22,82 @@ class ABCModelingEngine(object):
-            information.
-        """
-        pass
-
-    @abstractmethod
-    def add_particles(self, particles):
-        """Add particle container to the modeling engine
-
-        Parameters
-        ----------
-        particles: ABCParticles
-            particle container to be added.
-
-        Returns
-        -------
-        ABCParticles
-            A particle container to be used to update/query the internal
-            representation stored inside the modeling-engine. See
-            get_particles for more information.
-
-        """
-        pass
-
-    @abstractmethod
-    def delete_lattice(self, name):
-        """Delete a lattice
+    def remove_dataset(self, name):
+        """ Remove a dataset from the internal
 
         Parameters
         ----------
         name: str
-            name of lattice to be deleted
-
-        """
-        pass
-
-    @abstractmethod
-    def delete_mesh(self, name):
-        """Delete a mesh
+            name of CUDS container to be deleted
 
-        Parameters
-        ----------
-        name: str
-            name of mesh to be deleted
+        Raises
+        ------
+        ValueError:
+            If there is no dataset with the given name
 
         """
         pass
 
     @abstractmethod
-    def delete_particles(self, name):
-        """Delete a particle container
+    def get_dataset(self, name):
+        """ Get the dataset
 
         Parameters
         ----------
         name: str
-            name of particle container to be deleted
-
-        """
-        pass
-
-    @abstractmethod
-    def get_lattice(self, name):
-        """ Get lattice
-
-        The returned lattice can be used to query and update the state of the
-        lattice inside the modeling engine.
-
-        Returns
-        -------
-        ABCLattice
-
-        """
-        pass
-
-    @abstractmethod
-    def get_mesh(self, name):
-        """ Get mesh
-
-        The returned mesh can be used to query and update the state of the
-        mesh inside the modeling engine.
-
-        Returns
-        -------
-        ABCMesh
-
-        """
-        pass
-
-    @abstractmethod
-    def get_particles(self, name):
-        """ Get particle container
-
-        The returned particle container can be used to query and update the
-        state of the particle container inside the modeling engine.
+            name of CUDS container to be retrieved.
 
         Returns
         -------
-        ABCParticles
-
-        """
-        pass
-
-    @abstractmethod
-    def iter_lattices(self, names=None):
-        """ Returns an iterator over a subset or all of the lattices.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific lattices to be iterated over. If names is not
-            given, then all lattices will be iterated over.
+        container :
+            A proxy of the dataset named ``name`` that is stored
```

Comment:
suggestion: I would avoid using the word `proxy` in the case. I have seen a number of cases were developers can get into a heated discussion on terms like proxy and adapter and which pattern we follow. I would suggest something like

```
container:
    An object implementing the CUDS container API providing access to the information stored in the engine.
```

Related links: (http://programmers.stackexchange.com/questions/201397/difference-between-the-adapter-pattern-and-the-proxy-pattern).